pkgs/by-name/br/bruno/package.nix

```diff
@@ -1,61 +1,127 @@
 { lib
-, stdenv
-, fetchurl
-, autoPatchelfHook
-, dpkg
-, wrapGAppsHook
-, alsa-lib
-, gtk3
-, mesa
-, nspr
-, nss
-, systemd
+
+, fetchFromGitHub
+, buildNpmPackage
 , nix-update-script
+, electron-bin
+, writeShellScriptBin
+, makeWrapper
+, copyDesktopItems
+, makeDesktopItem
+, pkg-config
+, pixman
+, cairo
+, pango
+, npm-lockfile-fix
 }:
 
-stdenv.mkDerivation rec {
+let
+  electron = electron-bin;
+in
+
+buildNpmPackage rec {
   pname = "bruno";
-  version = "1.5.0";
+  version = "1.5.1";
 
-  src = fetchurl {
-    url = "https://github.com/usebruno/bruno/releases/download/v${version}/bruno_${version}_amd64_linux.deb";
-    hash = "sha256-ptrayWDnRXGUC/mgSnQ/8sIEdey+6uoa3LGBGPQYuY8=";
+  src = fetchFromGitHub {
+    owner = "usebruno";
+    repo = "bruno";
+    rev = "v${version}";
+    hash = "sha256-GgXnsPEUurPHrijf966x5ldp+1lDrgS1iBinU+EkdYU=b";
+
+    postFetch = ''
+      ${lib.getExe npm-lockfile-fix} $out/package-lock.json
+    '';
```

Comment:
I think we should rather do this in postPatch to make the download easier to reproduce.

Also that this does network requests is not that great. We should probably just apply a patch to get the hashes.

Reply:
Build complains about inconsistencies in package-lock.json.

That way the bump script already fixes the lock and updates the hash. As the lock fix is reproducible it's fine.

fetchFromGitHub doesn't have a patchPhase tho

Reply:
> As the lock fix is reproducible it's fine.

but we now depend on npmjs.com working to be able to fetch the src, right? If someone has a copy of the archive but the service is no longer available or some download is yanked, then the FOD cannot be produced anymore. A patch file shouldn't have that problem.

Reply:
We will still depend on npm to generate the node_modules to build the package. BTW for end users this bundle will be cached.